caikit/runtime/http_server/http_server.py

```diff
@@ -350,49 +388,154 @@ def _bind_routes(self, service: ServicePackage):
-        return request_params
+        # Bind POST to deploy a model
+        deploy_spec = MODEL_MANAGEMENT_SERVICE_SPEC["service"]["rpcs"][0]
+        assert deploy_spec["name"] == "DeployModel"
+        deploy_dataobject_request = DataBase.get_class_for_name(
+            deploy_spec["input_type"]
+        )
+        deploy_pydantic_request = dataobject_to_pydantic(deploy_dataobject_request)
+        deploy_dataobject_response = DataBase.get_class_for_name(
+            deploy_spec["output_type"]
+        )
+        deploy_pydantic_response = dataobject_to_pydantic(deploy_dataobject_response)
+
+        @self.app.post(
+            MODEL_MANAGEMENT_ENDPOINT,
+            responses=self._get_response_openapi(
+                deploy_dataobject_response, deploy_pydantic_response
+            ),
+            openapi_extra=self._get_request_openapi(deploy_pydantic_request),
+            response_class=Response,
+        )
+        async def _deploy_model(context: Request) -> Response:
+            """POST handler for deploying a model"""
+            try:
+                request = await pydantic_from_request(deploy_pydantic_request, context)
+                result = self.model_management_servicer.DeployModel(request, None)
+                return Response(
+                    content=deploy_dataobject_response.from_proto(result).to_json(),
+                    media_type="application/json",
+                )
+            except Exception as err:
+                if error_content := self._handle_exception(err):
+                    return Response(
+                        content=json.dumps(error_content),
+                        status_code=error_content["code"],
+                    )
+                raise
+
+        # Bind DELETE to undeploy a model
+        undeploy_spec = MODEL_MANAGEMENT_SERVICE_SPEC["service"]["rpcs"][1]
+        assert undeploy_spec["name"] == "UndeployModel"
+        undeploy_dataobject_request = DataBase.get_class_for_name(
+            undeploy_spec["input_type"]
+        )
+        undeploy_dataobject_response = DataBase.get_class_for_name(
+            undeploy_spec["output_type"]
+        )
+        undeploy_pydantic_response = dataobject_to_pydantic(
+            undeploy_dataobject_response
+        )
+
+        @self.app.delete(
+            MODEL_MANAGEMENT_ENDPOINT,
+            responses=self._get_response_openapi(
+                undeploy_dataobject_response, undeploy_pydantic_response
+            ),
+            response_class=Response,
+        )
+        def _undeploy_model(model_id: Annotated[str, Query]) -> Response:
+            """DELETE handler for undeploying a model"""
+            try:
+                result = self.model_management_servicer.UndeployModel(
+                    undeploy_dataobject_request(model_id).to_proto(), None
+                )
+                return Response(
+                    content=undeploy_dataobject_response.from_proto(result).to_json(),
```

Comment:
It looks like we're doing a lot of extra translation? The request is going:
`json -> data model -> protobuf ` and then the response is going
`data model -> protobuf -> data model -> json`

Probably not a performance concern since this is a deploy-time API, but certainly adds complexity and inflates the bug surface.

Any chance we can refactor that out? Maybe just call the model manager directly here and not involve protobuf at all?

Reply:
Yeah, this does kind of suck. I don't think there's a simpler way to do this without a _big_ refactor that puts `pydantic` at the heart of `dataobject` unfortunately. There's no guarantee that `json -> protobuf` or `protobuf -> json` will work since individual `data model` objects may have custom logic for to/from json/proto. In this specific case, we know they don't but I would be very unsurprised if it were added down the line to support special upload capabilities and I wouldn't want this to suddenly break.

Reply:
What I'm getting at here is a smaller refactor, not overhauling dataobjects. Instead of having this complexity around here to wrap the grpc servicers (which is generic and very cool and all), can't we just write the logic for the requests and responses inline since they're gonna be reasonably static?

```
def _undeploy_model(model_id: Annotated[str, Query]) -> Response:    
    self.model_manager.undeploy(model_id)
    return Response(content={"model_id": model_id})

def _deploy_model(context: Request) -> Response:
    request = await pydantic_from_request(deploy_pydantic_request, context)
    loaded_model = self._model_manager.deploy_model(
        model_id=request.model_id,
        model_files={f.filename: f.data for f in request.model_files},
        wait=False,
    )

    model_info = ModelInfo(
        model_path=loaded_model.path(),
        name=loaded_model.id(),
        size=loaded_model.size(),
        loaded=loaded_model.loaded(),
    )
    
    return Response(content=model_info.to_json())
```

Reply:
Ok, I dug in deeper and you were totally right! Much simpler now